lwtime/reltime.go

```diff
@@ -0,0 +1,211 @@
+			mondays = append(mondays, RelDate{year, month, day})
+		}
+		start = start.Add(d)
+	}
+	return mondays
+}
+
+type RelTimeUnit string
+
+const (
+	relTimeRE             = `^([+-])?(?:(\d+)(\w+))?(?:@(\w+))?$`
+	Year      RelTimeUnit = "y"
+	Month     RelTimeUnit = "mon"
+	Week      RelTimeUnit = "w"
+	Day       RelTimeUnit = "d"
+	Hour      RelTimeUnit = "h"
+	Minute    RelTimeUnit = "m"
+	Second    RelTimeUnit = "s"
+)
+
+func (rtu RelTimeUnit) IsValid() bool {
+	switch RelTimeUnit(strings.ToLower(string(rtu))) { // inline lowercase conversion
+	case Year, Month, Week, Day, Hour, Minute, Second, RelTimeUnit(""):
+		return true
+	}
+	return false
+}
+
+func (rtu RelTimeUnit) SnapTime(inTime time.Time) (outTime time.Time, err error) {
+	// immediately short circuit if snap is invalid
+	if !rtu.IsValid() {
+		err = errors.New(fmt.Sprintf(
+			"snap (%s) is not a valid relative time unit", rtu))
+		return
+	}
+
+	year, month, day := inTime.Date()
+	hour := inTime.Hour()
+	minute := inTime.Minute()
+	second := inTime.Second()
+	nano := inTime.Nanosecond()
+
+	switch RelTimeUnit(strings.ToLower(string(rtu))) {
+	case Week:
+		year, week := inTime.ISOWeek()
+		relDate := Mondays(year)[week-1]
+		outTime = time.Date(
+			relDate.Year,
+			relDate.Month,
+			relDate.Day,
+			0, 0, 0, 0, inTime.Location(),
+		)
+		return
+	case Year:
+		month = 1
+		fallthrough
+	case Month:
+		day = 1
+		fallthrough
+	case Day:
+		hour = 0
+		fallthrough
+	case Hour:
+		minute = 0
+		fallthrough
+	case Minute:
+		second = 0
+		fallthrough
+	case Second:
+		nano = 0
+	}
+	outTime = time.Date(year, month, day, hour, minute, second, nano, inTime.Location())
+	return
+}
+
+type RelTime struct {
+	Num  string
+	INum int
+	Unit RelTimeUnit
+	Snap RelTimeUnit
+}
+
+func (rt *RelTime) Parse(s string) error {
+	var rt_parts []string
+	// now is equivelant to +0s
+	if s == "now" {
+		s = "+0s"
+	}
+	// regex
+	re := regexp.MustCompile(relTimeRE)
+	if rt_parts = re.FindStringSubmatch(s); s == "" || rt_parts == nil {
+		return errors.New(fmt.Sprintf("relative time specifier (%s) is invalid", s))
+	}
+	// Num
+	if rt_parts[1] == "-" {
+		rt.Num = rt_parts[1] + rt_parts[2]
+	} else {
+		rt.Num = rt_parts[2]
+	}
+	var err error
+	rt.INum, err = strconv.Atoi(rt.Num)
+	if err != nil {
+		rt.Num = "0"
+		rt.INum = 0
+	}
+	// Unit
+	rt.Unit = RelTimeUnit(strings.ToLower(string(rt_parts[3])))
+	if !rt.Unit.IsValid() {
+		return errors.New(fmt.Sprintf("invalid unit for relative time specifier (%s)", s))
+	}
+	if rt.Unit == RelTimeUnit("") {
+		rt.Unit = Second
+	}
+	// Weeeeeeek
+	if rt.Unit == Week {
+		rt.INum = rt.INum * 7
+		rt.Num = strconv.Itoa(rt.INum)
+		rt.Unit = Day
+	}
+	// Snap
+	rt.Snap = RelTimeUnit(strings.ToLower(string(rt_parts[4])))
+	if !rt.Snap.IsValid() {
+		return errors.New(fmt.Sprintf("invalid snap for relative time specifier (%s)", s))
+	}
+	return nil
+}
+
+func (rt RelTime) Time() (outTime time.Time, err error) {
+	outTime, err = rt.Time_(time.Now())
+	return
+}
```

Comment:
unnecessary naked return
```suggestion
func (rt RelTime) Time() (time.Time, error) {
	return rt.Time_(time.Now())
}
```

Reply:
but what is this function for?

Reply:
Similar to `Range_` this function (`Time()`) calls the `Time_()` function with the current time.

Reply:
Fixed.